lib/scan/wp_xmlrpc/engine.py

```diff
@@ -0,0 +1,205 @@
+            random.choice(string.ascii_letters + string.digits) for _ in range(20))
+        __log_into_file(thread_tmp_filename, 'w', '1', language)
+        trying = 0
+        if target_type(target) != "HTTP":
+            target = 'https://' + target
+        for port in ports:
+            if test(str(target), port, retries, timeout_sec, headers,
+                    socks_proxy, verbose_level, trying, total_req, total, num, language) is True:
+                keyboard_interrupt_flag = False
+                t = threading.Thread(target=check,
+                                        args=(
+                                            target, port, headers, timeout_sec, log_in_file, language,
+                                            retries, time_sleep, thread_tmp_filename, socks_proxy,
+                                            scan_id, scan_cmd))
+                threads.append(t)
+                t.start()
+                trying += 1
+                if verbose_level > 3:
+                    info(messages(language, "trying_message").format(trying, total_req, num, total, target_to_host(target),
+                                                                    port, 'wp_xmlrpc_scan'))
+                while 1:
+                    try:
+                        if threading.activeCount() >= thread_number:
+                            time.sleep(0.01)
+                        else:
+                            break
+                    except KeyboardInterrupt:
+                        keyboard_interrupt_flag = True
+                        break
+                if keyboard_interrupt_flag:
+                    break
+
+            else:
+                warn(messages(language, "open_error").format(target))
+
+        # wait for threads
+
+        kill_switch = 0
+        kill_time = int(
+            timeout_sec / 0.1) if int(timeout_sec / 0.1) != 0 else 1
+        while 1:
+            time.sleep(0.1)
+            kill_switch += 1
+            try:
+                if threading.activeCount() == 1 or kill_switch == kill_time:
+                    break
+            except KeyboardInterrupt:
+                break
+        thread_write = int(open(thread_tmp_filename).read().rsplit()[0])
+        if thread_write == 1 and verbose_level != 0:
+
+            info(messages(language, "no_vulnerability_found").format(
+                'XML-RPC'))
+            data = json.dumps({'HOST': target, 'USERNAME': '', 'PASSWORD': '', 'PORT': port, 'TYPE': 'wp_xmlrpc_scan',
+                            'DESCRIPTION': messages(language, "no_vulnerability_found").format("XML-RPC DOS attacks"), 'TIME': now(), 'CATEGORY': "scan", 'SCAN_ID': scan_id, 'SCAN_CMD': scan_cmd}) + "\n"
+            __log_into_file(log_in_file, 'a', data, language)
+        os.remove(thread_tmp_filename)
+    else:
+        warn(messages(language, "input_target_error").format(
+            'wp_xmlrpc_scan', target))
```

Comment:
@aman566 please change to wp_xmlrpc_dos_vuln and scan this file with **pylint** and **flake8**